### PR TITLE
SDK - Components - Split load_component functions into loading the spec and creating task factory

### PR DESCRIPTION
**Pull request by Ark-kun**
The PR is a refactoring.
Split all load_component* methods in _components and _component_store into _load_component_spec* and creating task factory from that spec.
This makes it easier to load the spec without having to create task factory functions.

### CODE REVIEW

File: sdk/python/kfp/components/_components.py [hunk 1]
```diff
@@ -119,20 +115,52 @@ def load_component_from_text(text):
...
+    return _load_component_spec_from_yaml_or_zip_bytes(resp.content)
 
 
 _COMPONENT_FILE_NAME_IN_ARCHIVE = 'component.yaml'
 
 
-def _load_component_from_yaml_or_zip_bytes(bytes, component_filename=None, component_ref: ComponentReference = None):
+#def _load_component_from_yaml_or_zip_bytes(bytes, component_filename=None, component_ref: ComponentReference = None):
+#    component_spec = _load_component_spec_from_yaml_or_zip_bytes(bytes)
+#    return _create_task_factory_from_component_spec(component_spec, component_filename, component_ref)
```
Comment (chensun) [inline sdk/python/kfp/components/_components.py:152]: Do you still need these?

Comment (Ark-kun) [reply sdk/python/kfp/components/_components.py]: Fixed.

**Comment by Ark-kun** [comment]
/hold
In case Travis does not turn green

**Comment by Ark-kun** [comment]
/hold cancel
@Bobgy @numerology Can you please re-LGTM?

**Pull request merged**